### PR TITLE
feat: adds recently searched journeys to location search and ability to clear history. Closes #673 

### LINKED ISSUE #673: [Feature] Add last searched journeys
## Origin
_Links to received feedback, user research or other findings._

See issue #420

## Motivation
_A short description of what user needs or business goals this feature will solve._

Make it easier to search for frequently used journeys.

## Hypotheses and assumptions
_A list of hypotheses and assumptions we have made about the user or the proposed solution._

The user may want to reuse a previous journey search which is not relative to the current location.

## Proposed solution
_A coarse description of a proposed solution, that may include wireframes and graphic design._

When using the assistant to search for the departure-point, show a list of recently used complete (origin to destination) searches.

Figma / App 1.0 / Action / Search https://www.figma.com/file/WsvD8b5PnUwvvRJiyRe6I9/App-1.0?node-id=1839%3A7190

image

### PR DESCRIPTION
**Pull request by mikaelbr**
Closes #673

This PR adds the ability to re-use previous journey searches.

With this, I also added the ability to clear search history as a part of privacy. I feel this is more important now with actual journey searches, even though it is still only stored locally. (I also needed it for testing).

### Example screen shots

Screenshot 2021-03-17 at 22 47 10
Screenshot 2021-03-17 at 22 47 06t of
Screenshot 2021-03-17 at 22 46 46

### CODE REVIEW

**Comment by mikaelbr** [comment]
Limits the current latest journey suggestions to 3 for now. Do we think this needs to be configurable through remote config so we can update that on the fly? cc @pettersuul @hildeor

**Comment by gorandalum** [comment]
Nice. I'll take a look at this now.

Would it make sense to have these "Most recent journeys" on the Assistant screen also? I feel they kinda belong there more than the favourite places which are there now.

Screenshot 2021-03-19 at 08 25 36

**Review by gorandalum — CHANGES_REQUESTED** [review]
Great, this will be a nice new functionality for the users.

I feel that the code is kind of hard to follow, but maybe it is no way around it when we give LocationSearch this added functionality. One other hint that the code is complex is that it all looked ok when reviewing, but when testing I found one or more bugs (I don't know if it is the same one).

Bug 1:
I first selected a recent journey between Kvennhusveien 8 (My position) and Ila. After that I tried changing the from-location, but after selecting Solsiden the from-location was still Kvennhusveien 8:


Bug 2:
I first selected a recent journey between Ila and Solsiden. After that I changed the to-location to Hallset, and then the from-location was unintentionally changed to my position (Kvennhusveien 8):

**Comment by mikaelbr** [comment]
@gorandalum Simplified logic now and seems to fix the issue also. Check if the logic is easier to follow now.

**Comment by gorandalum** [comment]
LGTM! :+1:
And no insects found when testing.

**Review by gorandalum — APPROVED** [review]
